src/main/java/net/minecraftforge/fluids/FluidRegistry.java

```diff
@@ -281,14 +110,9 @@ public static boolean isUniversalBucketEnabled()
      */
     public static boolean addBucketForFluid(Fluid fluid)
     {
-        if(fluid == null) {
+        if(fluid == null || !ForgeRegistries.FLUIDS.containsValue(fluid)) {
```

Comment:
This `if` statement should be replaced with `com.google.common.base.Preconditions#checkNotNull` and `checkArgument`. The javadoc says it returns false if it was registered already or couldn't be registered, but realistically these two cases are completely invalid inputs and will be hard for devs to notice without a crash.